driver-mapping/src/test/java/com/datastax/driver/mapping/PropertyNamingStrategyTest.java

```diff
@@ -0,0 +1,314 @@
+                .setCassandraConvention(new CommonNamingConventions.LowerCamelCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<SnakeCaseToCamelCase> mapper = mappingManager.mapper(SnakeCaseToCamelCase.class);
+        assertThat(mapper.get(1).getMy_xml_parser()).isEqualTo(1);
+        assertThat(mapper.get(1).getOther()).isEqualTo(2);
+    }
+
+    @Table(name = "camel_case_table")
+    public static class SnakeCaseToCamelCase {
+
+        @PartitionKey
+        private int my_xml_parser;
+
+        private int other;
+
+        public int getMy_xml_parser() {
+            return my_xml_parser;
+        }
+
+        public void setMy_xml_parser(int my_xml_parser) {
+            this.my_xml_parser = my_xml_parser;
+        }
+
+        public int getOther() {
+            return other;
+        }
+
+        public void setOther(int other) {
+            this.other = other;
+        }
+
+    }
+
+    @Test(groups = "short")
+    public void test_overrides() {
+        MapperConfiguration conf = new MapperConfiguration();
+        conf.getNamingStrategy()
+                .setJavaConvention(new CommonNamingConventions.LowerSnakeCase())
+                .setCassandraConvention(new CommonNamingConventions.LowerCamelCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<TestOverrides> mapper = mappingManager.mapper(TestOverrides.class);
+        assertThat(mapper.get(1).getMyXmlParser()).isEqualTo(1);
+        assertThat(mapper.get(1).getGoo()).isEqualTo(2);
+    }
+
+    @Table(name = "camel_case_table")
+    public static class TestOverrides {
+
+        @PartitionKey
+        @Column(caseSensitive = true)
+        private int myXmlParser;
+
+        @Column(name = "other")
+        private int goo;
+
+        public int getMyXmlParser() {
+            return myXmlParser;
+        }
+
+        public void setMyXmlParser(int myXmlParser) {
+            this.myXmlParser = myXmlParser;
+        }
+
+        public int getGoo() {
+            return goo;
+        }
+
+        public void setGoo(int goo) {
+            this.goo = goo;
+        }
+
+    }@Test(groups = "short")
+
+    public void test_recommended_settings() {
+        MapperConfiguration conf = new MapperConfiguration();
+        conf.getNamingStrategy()
+                .setJavaConvention(new CommonNamingConventions.LowerCamelCase("m", "_"))
+                .setCassandraConvention(new CommonNamingConventions.LowerSnakeCase());
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<TestRecommendedSettings> mapper = mappingManager.mapper(TestRecommendedSettings.class);
+        assertThat(mapper.get(1).getmMyXmlParser()).isEqualTo(1);
+        assertThat(mapper.get(1).get_other()).isEqualTo(2);
+    }
+
+    @Table(name = "snake_case_table")
+    public static class TestRecommendedSettings {
+
+        @PartitionKey
+        private int mMyXmlParser;
```

Comment:
Sort of confused how this works.  Since the Java Convention is `LowerCamelCase` (`new CommonNamingConventions.LowerCamelCase("m", "_")`) should this be `mmyXmlParser`, as the first `m` would be stripped and `myXmlParser` is `LowerCamelCase` (vs `MyXmlParser` which would be `UpperCamelCase`)?